s3/src/main/scala/monix/connect/s3/MultipartUploadSubscriber.scala

```diff
@@ -0,0 +1,185 @@
+  * that finishes with a message to S3 that multipart upload is completed, in which the concatenation
+  * of each of the individual chunks will end up stored into the same object.
+  * Implement the functionality to perform an S3 multipart update from received chunks of data.
+  */
+private[s3] class MultipartUploadSubscriber(
+  bucket: String,
+  key: String,
+  minChunkSize: Int = MultipartUploadSubscriber.awsMinChunkSize,
+  acl: Option[String],
+  contentType: Option[String] = None,
+  grantFullControl: Option[String],
+  grantRead: Option[String],
+  grantReadACP: Option[String],
+  grantWriteACP: Option[String],
+  serverSideEncryption: Option[String] = None,
+  sseCustomerAlgorithm: Option[String] = None,
+  sseCustomerKey: Option[String] = None,
+  sseCustomerKeyMD5: Option[String] = None,
+  ssekmsEncryptionContext: Option[String],
+  ssekmsKeyId: Option[String],
+  requestPayer: Option[String])(
+  implicit
+  s3Client: S3AsyncClient)
+  extends Consumer.Sync[Array[Byte], CompleteMultipartUploadResponse] {
+
+  private val createRequest: CreateMultipartUploadRequest =
+    S3RequestBuilder.createMultipartUploadRequest(
+      bucket = bucket,
+      key = key,
+      contentType = contentType,
+      acl = acl,
+      grantFullControl = grantFullControl,
+      grantRead = grantRead,
+      grantReadACP = grantReadACP,
+      grantWriteACP = grantWriteACP,
+      requestPayer = requestPayer,
+      serverSideEncryption = serverSideEncryption,
+      sseCustomerAlgorithm = sseCustomerAlgorithm,
+      sseCustomerKey = sseCustomerKey,
+      sseCustomerKeyMD5 = sseCustomerKeyMD5,
+      ssekmsEncryptionContext = ssekmsEncryptionContext,
+      ssekmsKeyId = ssekmsKeyId
+    )
+
+  private val uploadId: Task[String] =
+    Task
+      .from(s3Client.createMultipartUpload(createRequest))
+      .map(_.uploadId())
+      .memoize
+
+  private var buffer: Array[Byte] = Array.emptyByteArray
+
+  private var completedParts: List[CompletedPart] = List.empty[CompletedPart]
+
+  private var partN = 1
+  def createSubscriber(
+    callback: Callback[Throwable, CompleteMultipartUploadResponse],
+    s: Scheduler): (Subscriber.Sync[Array[Byte]], AssignableCancelable) = {
+    val out = new Subscriber.Sync[Array[Byte]] {
+      implicit val scheduler = s
+
+      /**
+        *
+        * If the chunk lenght is bigger than the minimum size, perfom the part request [[UploadPartRequest]],
+        * which returns a [[CompletedPart]].
+        * In the case it is not, the chunk is added to the buffer waiting to be aggregated with the next one.
+        */
+      def onNext(chunk: Array[Byte]): Ack = {
+        if (chunk.length < minChunkSize) {
+          buffer = buffer ++ chunk
+        } else {
+          try {
+            uploadPart(bucket, key, partN, uploadId, buffer)
+          } catch {
+            case ex if NonFatal(ex) => onError(ex)
```

Comment:
I have removed that `try catch`  because now with the non sync subscriber. 
Just added `.onErrorFallbackTo(Task(Ack.Stop))` that would cancel the upload whenever there is a failure, but not sure if right now the error would correctly be returned or not...

Reply:
Changed that, now if there is an error during the the process it will stop. 
Will add a new issue to implement a `retry` logic before before to acknowledge the Stop.